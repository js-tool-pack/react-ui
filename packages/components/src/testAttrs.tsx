import { fireEvent, render } from '@testing-library/react';
import type { PropsBase } from '@pkg/shared';
import React from 'react';

export function testAttrs<
  T extends { attrs?: Partial<React.HTMLAttributes<HTMLElement>> },
>(component: React.FC<T>, displayName?: string) {
  const name = displayName || component.displayName || component.name;
  test(name ? `${name} attrs` : `attrs`, () => {
    const onClick = jest.fn();

    const FC = component as React.FC<PropsBase>;
    const { container } = render(
      <FC
        attrs={{
          style: { width: '20px' },
          className: 'foo',
          role: 'note',
          onClick,
        }}
      />,
    );

    expect(container.firstChild).toHaveClass('foo');
    expect(container.firstChild).toHaveAttribute('role', 'note');
    expect(container.firstChild).toHaveStyle({ width: '20px' });

    expect(onClick).not.toBeCalled();
    fireEvent.click(container.firstChild!);
    expect(onClick).toBeCalled();
  });
}
