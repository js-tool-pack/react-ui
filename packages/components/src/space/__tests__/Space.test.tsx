import { testAttrs } from '~/testAttrs';
import { Space } from '..';
import { render } from '@testing-library/react';

describe('Space', () => {
  testAttrs(Space);

  test('snap', () => {
    const { container } = render(
      <Space>
        <div>1</div>
        <div>2</div>
      </Space>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('tag', () => {
    const { container } = render(<Space tag="div" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('gap', () => {
    const { container } = render(<Space gap={20} />);
    expect(container.firstChild).toHaveStyle({ gap: '20px' });
  });

  test('vertical', () => {
    const { container } = render(<Space vertical />);
    expect(container.firstChild).toHaveClass('t-space--vertical');
  });

  test('inline', () => {
    const { container } = render(<Space inline />);
    expect(container.firstChild).toHaveClass('t-space--inline');
  });

  test('fill', () => {
    const { container } = render(<Space fill fillRatio={20} />);
    expect(container.firstChild).toHaveClass('t-space--fill');
    expect(container.firstChild).toHaveStyle({ '--t-space-fill-ratio': '20%' });
  });

  test('separator', () => {
    const { container } = render(
      <Space separator={<div className="sep">|</div>}>
        <div>1</div>
        <div>2</div>
      </Space>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
