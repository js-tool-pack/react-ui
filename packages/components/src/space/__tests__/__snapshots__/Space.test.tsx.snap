// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Space separator 1`] = `
<section
  class="t-space"
  style="--t-space-fill-ratio: 100%;"
>
  <div>
    1
  </div>
  <div
    class="sep t-space__separator"
  >
    |
  </div>
  <div>
    2
  </div>
</section>
`;

exports[`Space snap 1`] = `
<section
  class="t-space"
  style="--t-space-fill-ratio: 100%;"
>
  <div>
    1
  </div>
  <div>
    2
  </div>
</section>
`;

exports[`Space tag 1`] = `
<div
  class="t-space"
  style="--t-space-fill-ratio: 100%;"
/>
`;
